sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java

```diff
@@ -175,17 +180,28 @@ public FormRecognizerAsyncClient buildAsyncClient() {
     }
 
     private HttpPipeline getDefaultHttpPipeline(Configuration buildConfiguration) {
+        ClientOptions buildClientOptions = (clientOptions == null) ? new ClientOptions() : clientOptions;
```

Comment:
ClientOptions also allows users to add new headers. We should update our AddHeadersPolicy to include headers in ClientOptions.